### PR TITLE
Ft admin can add/view user group via api #159022582

### PR DESCRIPTION
**Pull request by precious-ijege**
### What does this PR do?

- It  allows admin user create and view **user groups** via the API

### Description of Task to be completed?

- An admin should be able to **POST** and **GET** _user-group_ via the API. Initially, the admin can only do that via the django UI, but this feature allows the admin to do so via the API.

- Ensure all tests are passing.

### How should this be manually tested?

- Visit any of the **user-group** endpoints.

### Screenshot
screen shot 2018-07-27 at 8 59 20 am
screen shot 2018-07-27 at 8 59 58 am

### CODE REVIEW

**Comment by precious-ijege** [comment]
@egichuri @collinmutembei I am still waiting for your review on this PR.

**Comment by collinmutembei** [comment]
@precious-ijege Peruse through the [best-practices](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices) repo and kindly refrain from going out of the recommended conventions. Update the PR title and description accordingly.

File: api/urls.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def __init__(self, trailing_slash='/?'):
 router.register('security-users', SecurityUserViewSet,
                 'security-users')
 router.register('asset-specs', AssetSpecsViewSet, 'asset-specs')
+router.register('user-group', GroupViewSet, 'user-group')
```
Comment (collinmutembei) [inline api/urls.py]: Lets call the route `user-groups`.

**Comment by collinmutembei** [comment]
@egichuri It would be best to enforce the case-insensitive uniqueness on this model as well. Probably by inheriting and overriding the model field on `django.contrib.auth.models.Group`.

screen shot 2018-07-31 at 10 52 54

**Comment by precious-ijege** [comment]
@egichuri @collinmutembei PR updated.

File: api/views.py [hunk 2]
```diff
@@ -282,6 +284,14 @@ class AssetSpecsViewSet(ModelViewSet):
     http_method_names = ['get', 'post', 'put']
 
 
+class GroupViewSet(ModelViewSet):
```
Comment (collinmutembei) [inline api/views.py]: Update the **GroupViewset** to format data before saving it to ensure we do not have case-insensitive duplicates as below.


screen shot 2018-08-01 at 13 33 53

Use a similar statement to this one 👇 
https://github.com/AndelaOSP/art-backend/blob/7d703b681db684b65a8a61620a842917ed706a0e/core/models/officeblock.py#L10

File: api/views.py [hunk 3]
```diff
@@ -286,6 +287,18 @@ class AssetSpecsViewSet(ModelViewSet):
...
+    serializer_class = UserGroupSerializer
+    queryset = Group.objects.all()
+    permission_classes = [IsAuthenticated, IsAdminUser]
+    authentication_classes = (FirebaseTokenAuthentication,)
+    http_method_names = ['get', 'post']
+
+    def perform_create(self, serializer):
+        name = " ".join(serializer.validated_data.get('name').title().split())
```
Comment (collinmutembei) [inline api/views.py]: screen shot 2018-08-03 at 11 30 23

Add a try catch statement to resolve this **IntegrityError**.

**Pull request merged**